src/main/java/ch/heigvd/pro/b04/auth/LoginController.java

```diff
@@ -28,14 +32,16 @@ public LoginController(ModeratorRepository moderators) {
    */
   @RequestMapping(value = "auth", method = RequestMethod.POST)
   @ResponseBody
-  public TokenCredentials login(@RequestBody UserCredentials credentials)
+  public String login(@RequestBody UserCredentials credentials)
       throws UnknownUserCredentialsException {
     // FIXME : Use proper token-based authentication, rather than returning the password of the
     //         user.
     Optional<Moderator> moderator = moderators.findByUsername(credentials.getUsername());
-    Optional<TokenCredentials> response = moderator.flatMap(m -> {
-      if (m.getSecret().equals(credentials.getPassword())) {
-        return Optional.of(TokenCredentials.builder().token(m.getSecret()).build());
+    Optional<String> response = moderator.flatMap(m -> {
+      if (m.getSecret().equals(
+          Hashing.sha512().hashString(credentials.getPassword() + Constants.HASH,
+              StandardCharsets.UTF_8).toString())) {
+        return Optional.of(TokenCredentials.builder().token(m.getSecret()).build().getToken());
```

Comment:
Do not roll your own crypto :D

Reply:
It's Java Guava librabry

Reply:
That's not what he means. What you're doing here could easily be flawed, and that's alright, you're not a security expert. But this is why we should use something that does the whole procedure for us. Can you look into Spring Security ?

Small example of where your code should be different:
You have stored one hash that you are using for all passwords, meaning that the same password will have the same hashed value. If our database was to be leaked, an attack using Rainbow Tables will work quite easily. On the contrary, if you randomly create a new salt for every password and store it in the database, you do not have that problem anymore, because every password (even if it is the same) will have a different hashed value because the salt changed!

I'm sure there are many other examples that we're not aware of, this is exactly why we should delegate this to a framework/library/whatever that can actually do it correctly.